packages/core/lib/process-request.ts

```diff
@@ -210,23 +232,35 @@ export const processRequest = async <TContext = {}, TRootValue = {}>(
           // execution result.
           if (isAsyncIterable(result)) {
             const asyncIterator = result[Symbol.asyncIterator]();
+            const transformedIterator = {
+              ...asyncIterator,
+              next: async () => {
+                const iteratorResult = await asyncIterator.next();
+                if (iteratorResult.done) {
+                  return iteratorResult;
+                }
+                return formatPayload({
+                  payload: iteratorResult.value,
+                  context,
+                  rootValue,
+                  document,
+                  operation,
+                })
+              },
+            };
```

Comment:
I wonder why you choose this over the following

```ts
const transformedIterator = (async function* () {
  for await (const payload of result) {
    yield formatPayload({
      payload,
      context,
      rootValue,
      document,
      operation,
    });
  }
})();
```

Reply:
Because i still want to respect throw, return and the rest